### DISCUSSION TITLE
chain.id is a number instead of bigint

### DISCUSSION THREAD

**Discussion by topocount** [wevm/viem#2735] [Question] [answered]
### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/viem/issues) for the bug I encountered.

### Viem Version

2.9.27

### Current Behavior

the chain id is typed as a `number`.

### Expected Behavior

the chainId is [specified as a uint256](https://eips.ethereum.org/EIPS/eip-1344) and should therefore be a `bigint`. I understand that it's unlikely a developer will create a chainId large enough require uint256, but I'm mostly curious what the reasoning here is, or if there is a specification that I'm missing. handling the up and downcasts when passing chainIds around as params is not a huge deal in and of itself but if I'm creating a higher level sdk I'd like to understand how this is intended to be handled.

**Comment by jxom** [comment]
There is no reasoning for this.

**Reply by frangio** [reply, marked as answer]
This is not correct. This change was already accepted in https://github.com/wevm/abitype/pull/257 where examples were given of chain ids that are too large for Number.
